repair/repair.cc

```diff
@@ -670,6 +679,7 @@ repair_info::repair_info(seastar::sharded<database>& db_,
     const sstring& keyspace_,
     const dht::token_range_vector& ranges_,
     const std::vector<sstring>& cfs_,
+    std::vector<utils::UUID> table_ids_,
     int id_,
```

Comment:
We should only accept the UUID:s, and let the api level do the conversion, this guarantees it is done only once and there is no inconsistency.

Reply:
We always pass the tables_ids and the tables_name(cfs) to repair_info converted from the api level. It is done once. I see no consistency problem. 

```
auto table_ids = get_table_ids(db.local(), keyspace, cfs)
repair_info(..., tables_ids, cfs)
```

Even if we only pass tables_ids to repair_info, it is much more convenient to store a list of names, we access the table names a lot. So, with you suggestion, we would do

```
auto table_ids = get_table_ids(db.local(), keyspace, cfs)
repair_info(..., tables_ids) {
     cfs = get_table_names_from_table_ids(table_ids)
}
```
It makes little sense to convert it back and forth.

Reply:
Are you two in agreement here? It's hard to tell

Reply:
I am not sure neither. Let's wait for avi to comment.

Reply:
I still don't understand it. uuids and table names can drift out of sync when a table is dropped and re-created. It is dangerous to use a table name twice, since the second time might refer to a different table. There is only one safe thing you can do with a table name - convert it to a uuid.

Why do you need the table names?

Reply:
It is used for logging in multiple places to tell which tables we are working on. After this patch,we do not use table name to find a table object any more.

Reply:
I agree it is better to pass it in one way. The downside is we need to convert it back and forth. To make progress, I pushed v4 which passes only the table id to repair_info and repair_info does the convert from table_ids to table_names again.